game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -375,4 +398,50 @@ public void changeChanceDecrementOrIncrementOnSuccessOrFailure(final IDelegateBr
       delegateBridge.addChange(ChangeFactory.attachmentPropertyChange(this, newChance, CHANCE));
     }
   }
+
+  private static Map<String, Function<IAttachment, AttachmentProperty<?>>> createPropertyMap() {
```

Comment:
I'm really concerned by the `createPropertyMap` pattern. Specifically:
- `Map<String, Function<IAttachment, AttachmentProperty<?>>>` indicates an abstraction problem, it's a complex map; pretty clearly complex enough it deserves to be a 1st order data type.
- `private static` this data is really 'attached' to the attachment. Seems to be a SRP violation.
- magic strings with poor cohesion scattered in the code, eg: `"conditions",` ideally those would be an enum, but at minimum consolidated. Going that route we can consolidate more of this configuration code, that will keep it easier to find and maintain.
- current structure is not readily testable, does not seem to be long term path.